numerapi/utils.py

```diff
@@ -34,15 +34,32 @@ def replace(dictionary, key, function):
 
 
 def download_file(url, dest_path, show_progress_bars=True):
+    file_size = 0
     r = requests.get(url, stream=True)
     r.raise_for_status()
     # Total size in bytes.
     total_size = int(r.headers.get('content-length', 0))
-
+    
+    if os.path.exists(dest_path):
+        file_size = os.stat(dest_path).st_size # File size in bytes
+        if file_size < total_size:
+            resume_header = {'Range': 'bytes=%d-' % file_size}
+            r = requests.get(url, headers=resume_header, stream=True,  verify=False, allow_redirects=True)
+        elif file_size == total_size:
+            # Download complete
+            return
+        else:
+            # Error, delete file and restart download
+            os.remove(dest_path)
+            file_size = 0
+            pass
+    
     # write dataset to file and show progress bar
     pbar = tqdm.tqdm(total=total_size, unit='B', unit_scale=True,
                      desc=dest_path, disable=not show_progress_bars)
-    with open(dest_path, "wb") as f:
+    for i in range(file_size//1024):
```

Comment:
What's the reason for that for loop? You could just forward the progress bar once. Also, please leave a comment here, took me a bit to understand what you are doing

Reply:
So normally, the progress bar would start off at 0% and get to a 100%. But in the event that say 30% of the file is already downloaded, the progress bar needs to initially start from 30% and get to 100%. The loop here sets the progress to reflect how much of the file is already downloaded. Now that I think of it, it can be simplified to 

```python
pbar.update(file_size)
```